packages/react/ds/src/table/table-header.tsx

```diff
@@ -74,10 +83,10 @@ export function TableHeader({
         className,
         { 'gi-w-12': !isChildrenString },
       )}
-      role={onSort ? 'button' : undefined}
-      data-sorted={!!onSort}
+      role={role}
+      data-sorted={!!sorted}
       data-header-string={isChildrenString}
-      tabIndex={onSort && isChildrenString ? 0 : -1}
+      tabIndex={sorted && isChildrenString ? 0 : -1}
```

Comment:
I believe you should keep: 
```tabIndex={onSort && isChildrenString ? 0 : -1}```

sorted could be false so we show the 'swap_vert' icon but still is focusable.